swap/swap.go

```diff
@@ -317,23 +348,27 @@ func verifyChequeAgainstLast(cheque *Cheque, lastCheque *Cheque, expectedAmount
 func (s *Swap) updateBalance(peer enode.ID, amount int64) (int64, error) {
 	//adjust the balance
 	//if amount is negative, it will decrease, otherwise increase
-	s.balances[peer] += amount
+	balance, exists := s.getBalance(peer)
```

Comment:
@holisticode this operation is not atomic. you should consider having a lock over this entire operation (maybe not the map lock, but the swap lock)

Reply:
@acud you are of course right basically.

However, `updateBalance` is not exported and only called from other functions which are locked (e.g. `Add`). 

`go` does not support recursive mutexes, and I think I share the view expressed here https://stackoverflow.com/a/14671462/169252 that it is not a good idea at all. 

The alternative would have to be to have even more additional locks for internal functions. This would make sure that such functions would be safe if called from somewhere else in the future. In the current flow, this would mean we additionally lock something which is already locked above in the stack. Finally, we would have to have single locks for `updateBalance`, `loadLastCheque` and `loadBalance`. I think this is impractical and ugly at best.

So the suggestion is to apply the patter described in the stackoverflow link above, which is to add a comment to this function that it needs to be called with the lock already held.

Reply:
added relevant comments to the `loadBalance`, `updateBalance` and `loadLastSentCheque` signatures.

@acud please let us know if you disagree with this approach.

Reply:
@acud I will merge this PR onto incentives, please comment there if anything. Want to avoid further delays to the master PR due to our review lags.

You can of course use this PR as reference to identify what we did.